Frontend/src/Modal.svelte

```diff
@@ -1,9 +1,22 @@
 <script>
-	export let entity //to UserPage
-	export let showModal = false; //to UserPage
+	import AddGroupForm from "./lib/forms/AddGroupForm.svelte";
+	import AddLocationForm from "./lib/forms/AddLocationForm.svelte";
+	import AddUserForm from './lib/forms/AddUserForm.svelte';
+
+	export let showModal = false; 
+	export let selectedManager;
+
 
 	let dialog; // HTMLDialogElement
 	$: if (dialog && showModal) dialog.showModal();
+
+	let currentHeader;
+	
+	let forms = {
+		"Users" : AddUserForm,  
+		"Groups" : AddGroupForm,
+		"Locations & Access Points" : AddLocationForm
+	};
```

Comment:
What if instead of:

`export let selectedManager;`
...
```
let forms = {
  "Users" : AddUserForm,  
  "Groups" : AddGroupForm,
  "Locations & Access Points" : AddLocationForm
};
```
...
`<svelte:component this={forms[selectedManager]} bind:header={currentHeader}/>`


You did:
`export let selectedManager;`

`<svelte:component this={selectedManager} bind:header={currentHeader}/>`

where you pass in your form.svelte of choice from the caller?

Reply:
Or better yet after looking at the full files, why not just replace 
`<svelte:component this={selectedManager} bind:header={currentHeader}/>` with `<slot>`
and then use Modal like:

```
<Modal bind:showModal={showGroupFormModal}>
   <AddGroupForm/>
</Modal
```

Reply:
I'm giving this approach a look now

Reply:
*We had our meeting and this will be addressed in: [https://github.com/WallochMatt/CertifID/tree/feature-CID-8](url)